test/e2e_node/log_path_test.go

```diff
@@ -0,0 +1,94 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e_node
+
+import (
+	"io/ioutil"
+	"strings"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/util/uuid"
+	"k8s.io/kubernetes/test/e2e/framework"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = framework.KubeDescribe("ContainerLogPath", func() {
+	f := framework.NewDefaultFramework("kubelet-container-log-path")
+	Describe("Pod containers", func() {
+		Context("with log printed to stdout", func() {
+			var ns, logDir string
+			BeforeEach(func() {
+				ns = f.Namespace.Name
+				logDir = "/var/log/containers/"
+			})
+			It("Correct log path on the node should have been created", func() {
+				podName := "log-pod" + string(uuid.NewUUID())
+				logContName := "log-container" + string(uuid.NewUUID())
+				// The expected container's log file name on host is:
+				// <pod_name>_<pod_namespace>_<container_name>-<container_id>.log
+				logPathPrefix := podName + "_" + ns + "_" + logContName + "-"
+
+				pod := &api.Pod{
+					ObjectMeta: api.ObjectMeta{
+						Name:      podName,
+						Namespace: ns,
+					},
+					Spec: api.PodSpec{
+						Containers: []api.Container{
+							{
+								Image:   "gcr.io/google_containers/busybox:1.24",
+								Name:    logContName,
+								Command: []string{"sh", "-c", "while true; do echo \"$(hostname)\"; sleep 1; done"},
+							},
+						},
+					},
+				}
+
+				pod, err := f.Client.Pods(ns).Create(pod)
+				if err != nil {
+					framework.Failf("unable to create pod %v: %v", podName, err)
+				}
+
+				err = f.WaitForPodRunning(podName)
+				Expect(err).NotTo(HaveOccurred(), "Failed waiting for pod %s to enter running state", podName)
+
+				files, err := ioutil.ReadDir(logDir)
```

Comment:
I mentioned that in the description, docker volume can not deal with linked files, so we can see the log files in Pods, but we cannot open it. You can reproduce it by `docker run -v "dir contains linked files"`